src/xc_integrator/local_work_driver/device/cuda/kernels/cuda_ssf_1d.cu

```diff
@@ -139,4 +139,227 @@ void partition_weights_ssf_1d( int32_t npts, int32_t natoms, const double* RAB,
+  const double*                           points_z,
+  const double*                           dist_scratch,
+        int32_t                           lddist,
+  const int32_t*                          iparent_device,
+  const double*                           dist_nearest_device,
+  const double*       __restrict__        w_times_f_device,
+        double*       __restrict__        exc_grad_w_device
+) {
+
+  // Frisch partition functions
+  auto gFrisch = [](double x) {
+
+    const double s_x  = x  * 1.5625; // / integrator::magic_ssf_factor<>;
+    const double s_x2 = s_x  * s_x;
+    const double s_x3 = s_x  * s_x2;
+    const double s_x5 = s_x3 * s_x2;
+    const double s_x7 = s_x5 * s_x2;
+
+    return ((35.)*(s_x - s_x3) + (21.)*s_x5 - (5.)*s_x7);
+  };
+  
+  auto sFrisch = [&] (double x) {
+    if( fabs(x) < integrator::magic_ssf_factor<> ) return (0.5 - (0.5/16.) * gFrisch(x));
+    else if( x >= integrator::magic_ssf_factor<> ) return 0.;
+    else                               return 1.;
+  };
+  
+  auto tFrisch = [&](double x) {
+    const double s_x  = x * 1.5625; // / integrator::magic_ssf_factor<>;
+    const double s_x2 = s_x  * s_x;
+    const double s_x3 = s_x  * s_x2;
+    const double numerator = (35.) * (s_x3 + (3.) * s_x2 + (3.) * s_x + (1.));
+    const double denominator = (x - integrator::magic_ssf_factor<>) * ((5.)*s_x3 + (20.)*s_x2 + (29.)*s_x + (16.));
+    return numerator / denominator ;
+  };
+
+  constexpr double safe_magic_ssf_bound = integrator::magic_ssf_factor<> - 1e-4;
+  constexpr double weight_tol = integrator::ssf_weight_tol;
+  constexpr double w_times_f_thresh = 1.e-12;
+
+  const int tid_x = threadIdx.x + blockIdx.x * blockDim.x;
+  const int nt_x  = blockDim.x  * gridDim.x;
+
+  for( int ipt = tid_x; ipt < npts; ipt += nt_x ) {
+
+    const auto w_times_f_i = w_times_f_device[ipt];
+    if (fabs(w_times_f_i) < w_times_f_thresh) continue; // weight derivative = 0 when p_A = 0
+    const auto iParent = iparent_device[ipt];
+
+    double sum = 0.; 
+    double parent_weight = 0.;
+
+    const double* const local_dist_scratch = dist_scratch + ipt * lddist;
+    const double dist_cutoff = 0.18 * dist_nearest_device[ipt]; // 0.5 * (1-integrator::magic_ssf_factor<>) * task.dist_nearest
+    if( local_dist_scratch[iParent] < dist_cutoff ) continue; //weight derivative = 0 when p_A = 1
+
+    // Do iParent First
+    {
+      const double ri = local_dist_scratch[ iParent ];
+      const double* const local_rab = RAB + iParent * ldRAB;
+
+      parent_weight = 1.;
+      for( int jCenter = 0; jCenter < natoms; jCenter++ ) 
+      if( parent_weight > weight_tol ) {
+      if( iParent != jCenter ) {
+      
+        const double rj = local_dist_scratch[ jCenter ];
+
+        const double mu = (ri - rj) * local_rab[ jCenter ]; // XXX: RAB is symmetric
+        parent_weight *= sFrisch( mu );
+
+      }
+      } else break;
+
+      sum += parent_weight;
+    }
+
+    // caculate sum
```

Comment:
typo

Reply:
This PR is meant to serve as a checkpoint for RC/1.0. Please direct this to an Issue.